src/Stratis.Bitcoin/Base/BackgroundWork/BackgroundWorkProvider.InnerClasses.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Signals;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Base.BackgroundWork
+{
+    public partial class BackgroundWorkProvider : IBackgroundWorkProvider
```

Comment:
I tend to prefer the name `AsyncProvider`
I suggest we create a new folder Called AsyncWork and put all this components there (move them out of `Utilities`).

Reply:
fine with me